firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -48,8 +48,8 @@ static CrossProcessLock acquire(Context appContext, String lockName) {
       File file = new File(appContext.getFilesDir(), lockName);
       channel = new RandomAccessFile(file, "rw").getChannel();
       // Use the file channel to create a lock on the file.
-      // This method blocks until it can retrieve the lock.
-      lock = channel.lock();
+      // This method does not block and only attempts to acquire the lock.
+      lock = channel.tryLock();
```

Comment:
hm iiuc using `tryLock()` here is equivalent to not using locks at all, i.e. it does not given any multiprocess synchronization at all, what is the reasoning behind it?

Reply:
I understand. Ideally I want to use lock() with timeout. File Channel doesnt have any methods with explicit timeouts. WDYT about putting the thread to sleep here explicitly?

Reply:
I guess it could work, although (given my comment below) I would suggest you first try to repro by writing multiprocess tests without a fix below and see if you can reproduce the issue, then if it turns out to be the root cause, see if promptly closing the fd fixes the issue.

Reply:
I see. Makes sesnse. I will try that. For now reverted back the lock and closed the randomaccessfile fd.

Reply:
also, have you seen the thread dumps from the IllegalStateException issue? it looks like it is not related to filelocks